custom_components/tahoma/light.py

```diff
@@ -144,10 +144,13 @@ def update(self):
         states = self.tahoma_device.active_states
 
         self._state = states.get(CORE_ON_OFF_STATE) == STATE_ON
-        self._brightness = states.get(CORE_LIGHT_INTENSITY_STATE)
 
-        self._hs_color = color_util.color_RGB_to_hs(
-            states.get(CORE_RED_COLOR_INTENSITY_STATE),
-            states.get(CORE_GREEN_COLOR_INTENSITY_STATE),
-            states.get(CORE_BLUE_COLOR_INTENSITY_STATE),
-        )
+        if CORE_LIGHT_INTENSITY_STATE in states:
+            self._brightness = states.get(CORE_LIGHT_INTENSITY_STATE)
+
+        if CORE_RED_COLOR_INTENSITY_STATE in states:
```

Comment:
Good spot, however I would check for all 3 values, since none of them can return None.

Reply:
```
[r, g, b] = [
            states.get(CORE_RED_COLOR_INTENSITY_STATE),
            states.get(CORE_GREEN_COLOR_INTENSITY_STATE),
            states.get(CORE_BLUE_COLOR_INTENSITY_STATE),
        ]
self._hs_color = (
            None if None in [r, g, b] else color_util.color_RGB_to_hs(r, g, b)
        )
```
this is better I think

Reply:
I also removed the test from brightness.